### PR TITLE
lammps: enable default packages

### PR DESCRIPTION
**Pull request by vishalkenchan**
Currently there is no way to enable default packages.

1. Could we enable default packages in lammps by default? Building these packages doesn't really take long.
2. Could we enable python embedding/interface whenever `BUILD_SHARED_LIBS` is `ON`? The use case of lammps' shared library typically involves using it with python interface.

`DEFAULT_PACKAGES` include:
```
ASPHERE BODY CLASS2 COLLOID COMPRESS CORESHELL DIPOLE
GRANULAR KSPACE MANYBODY MC MEAM MISC MOLECULE PERI QEQ REAX REPLICA
RIGID SHOCK SNAP SRD
```
cmake variable `ENABLE_ALL` enables all default packages.
(manybody is part of `DEFAULT_PACKAGES`)
@junghans

### CODE REVIEW

**Comment by junghans** [comment]
Agreed! I don't like `ALL`, it is just confusing. Plus `ALL` doesn't map on to any option in the makefile build system!

And for consistency, `PYTHON` should be its own variant, which conflicts with `~lib`. As an example, `parsplice` needs `lib`, but not `PYTHON`.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -80,7 +81,10 @@ def cmake_args(self):
             '-DBUILD_SHARED_LIBS={0}'.format(
                 'ON' if '+lib' in spec else 'OFF'),
             '-DENABLE_MPI={0}'.format(
-                'ON' if '+mpi' in spec else 'OFF')
+                'ON' if '+mpi' in spec else 'OFF'),
+            '-DENABLE_PYTHON={0}'.format(
```
Comment (junghans) [inline var/spack/repos/builtin/packages/lammps/package.py]: Make an own variant!

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 2]
```diff
@@ -80,7 +81,10 @@ def cmake_args(self):
...
+                'ON' if '+lib' in spec else 'OFF'),
+            '-DENABLE_ALL=ON'
         ]
```
Comment (junghans) [inline var/spack/repos/builtin/packages/lammps/package.py]: Some package in `ALL` conflicts with other, so no!

**Comment by alalazo** [comment]
@vishalkenchan I just changed the title to make it evident that this PR deals with just a package

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 3]
```diff
@@ -45,8 +45,12 @@ def url_for_version(self, version):
         return "https://github.com/lammps/lammps/archive/patch_{0}.tar.gz".format(
             vdate.strftime("%d%b%Y").lstrip('0'))
 
-    supported_packages = ['voronoi', 'rigid', 'user-netcdf', 'kspace',
-                          'latte', 'user-atc', 'user-omp', 'meam', 'manybody']
+    supported_packages = ['asphere', 'body', 'class2', 'colloid', 'compress',
+                          'coreshell', 'dipole', 'granular', 'kspace', 'latte',
+                          'manybody', 'mc', 'meam', 'misc', 'molecule', 'peri',
+                          'qeq', 'reax', 'replica', 'rigid', 'shock', 'snap',
+                          'srd', 'user-netcdf', 'user-atc', 'user-omp',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/lammps/package.py]: `user-atc` in front of `user-netcdf`.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 4]
```diff
@@ -55,6 +59,8 @@ def url_for_version(self, version):
             description='Build the liblammps in addition to the executable')
     variant('mpi', default=True,
             description='Build with mpi')
+    variant('python', default=True,
+            description='Build lammps shared library with python interface')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/lammps/package.py]: Could you just throw `python` into `supported_packages` as well?

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 5]
```diff
@@ -65,8 +71,10 @@ def url_for_version(self, version):
     depends_on('latte', when='+latte')
     depends_on('blas', when='+latte')
     depends_on('lapack', when='+latte')
+    depends_on('python', when='+python')
 
     conflicts('+latte', when='@:20170921')
+    conflicts('+python', when='~lib')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/lammps/package.py]: Can you make sure you have the dependencies of packages in here, too:
 https://github.com/lammps/lammps/blob/master/cmake/CMakeLists.txt#L123

Plus the `MPIIO` doesn't work without `MPI`.

**Comment by junghans** [comment]
There is a flake8 error, which needs to be fixed!

**Comment by vishalkenchan** [comment]
I guess this is ready to merge(?)

**Pull request merged**